### PR TITLE
Case insensitive DTag

### PR DESCRIPTION
**Pull request by bragaz**
## Description
This PR makes the `profiles` DTag case insensitive.
#492.
 
## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [ ] Wrote integration tests (simulation & CLI). 
- [ ] Updated the documentation. 
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by RiccardoM** [comment]
@bragaz Sims tests are failing, better check out why

**Review by RiccardoM — COMMENTED** [review]
Looks great. Can you just add a unit test to make sure that trying to save a profile with a specific DTag does not care about casing? What I mean is: 

1. Store an existing profile with DTag `test` and try storing a new profile from another user with DTag `Test`. This should error. 
2. Store an existing profile with DTag `test` and try updating it using DTag `Test` using the same account. This should not error and update the data correctly.

File: x/profiles/simulation/operations_dtag_transfers.go [hunk 1]
```diff
@@ -120,7 +120,10 @@ func randomDTagRequestTransferFields(
 	if err != nil {
 		return simtypes.Account{}, types.DTagTransferRequest{}, true
 	}
-	_ = k.StoreProfile(ctx, receiverProfile)
+	err = k.StoreProfile(ctx, receiverProfile)
+	if err != nil {
+		return simtypes.Account{}, types.DTagTransferRequest{}, true
+	}
```
Comment (RiccardoM) [inline x/profiles/simulation/operations_dtag_transfers.go:126]: I think this should not have been edited. I checked the failing sim test before editing this and the error was: 
```
failed to execute message; message index: 0: the request recipient does not have a profile yet: invalid request [/home/riccardo/Coding/Forbole/Desmos/x/profiles/keeper/msg_server_dtag_transfers.go:27]
        Comment: MsgRequestDTagTransfer
```

This probably means that something about storing or getting a profile using a DTag is not working correctly with casing. We should add more unit tests to see why this is not working.

**Comment by bragaz** [comment]
@RiccardoM I've added two more tests as you asked to ensure the fact that DTag casing is working properly and check also the error of the sim tests debugging it:
Schermata 2021-06-18 alle 10 03 20

Considering the error I think it's worth to add that check on `StoreProfile` right?

**Comment by RiccardoM** [comment]
> Considering the error I think it's worth to add that check on `StoreProfile` right?

I think this error started showing up now because case insensitive DTags actually work. If we take a look at the test code, what we do is: 
1. We get the user address and see if it has a profile 
2. If it does not have a profile, we store a new one with a random DTag

Previously, if the random DTag did not match only because of casing, nothing would happen. But now, since we have case insensitive DTags, this means that if the random DTag is equals to the one of another profile that error will show up. 

For this reason, I think we should review how simulation tests deal with profiles. Instead of getting a random recipient, and then creating a profile for such recipient, what we can do instead is getting a random profile. This way we do not have to generate a new one cause we will always use an already existing one. The only thing we need to check is to make sure that the recipient and message signer are not the same user.

**Comment by RiccardoM** [comment]
@bragaz Can you please add a CHANGELOG entry since this change is going to break the consensus state please? Then we're ready to merge this

**Comment by bragaz** [comment]
@RiccardoM done

**Review by RiccardoM — APPROVED** [review]

**Pull request merged**